android/app/build.gradle

```diff
@@ -348,6 +350,7 @@ if (isNewArchitectureEnabled()) {
 
 dependencyLocking {
     lockAllConfigurations()
+    lockMode = LockMode.STRICT
```

Comment:
Default is not strict and I'm thinking of removing it. The default behavior is described as follows:

> A successful build indicates that the same dependencies are used as stored in the lock state, regardless if new versions matching the dynamic selector have been produced.
> 
> The complete validation is as:
> - Existing entries in the lock state must be matched in the build
>   - A version mismatch or missing resolved module causes a build failure
> - Resolution result must not contain extra dependencies compared to the lock state

That sounds okay to me?

--- 

**Running in strict mode:**

When running `./gradlew generateLock saveLock` I am getting (correct) warnings about configurations that lack locking specifications. That is because we'd need each configuration to list a lock state.


Here is one for `':app:androidApis`

```
* What went wrong:
Execution failed for task ':app:generateLock'.
> Could not resolve all dependencies for configuration ':app:androidApis'.
   > Locking strict mode: Configuration ':app:androidApis' is locked but does not have lock state.
```
or another failure for `:app:lintChecks`

```
Could not resolve all dependencies for configuration ':app:lintChecks'.
Locking strict mode: Configuration ':app:lintChecks' is locked but does not have lock state.
```